src/replica.rs

```diff
@@ -15,26 +22,45 @@ use state::{ReplicaState, LeaderState, CandidateState, FollowerState};
 use state_machine::StateMachine;
 use store::Store;
 
-#[derive(Debug, Eq, PartialEq)]
-pub enum EmitType {
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum Timeout {
+    Election,
+    Heartbeat(ServerId),
+}
 
-    /// Do not emit the message.
-    None,
+pub struct Actions {
```

Comment:
Ok, interesting. I think this is good, it might need to be expanded a bit but we can do that later!